### ISSUE TITLE
Request to rename the terminal method on the LRA annotation

### ISSUE THREAD

**Issue by mmusgrov** [eclipse/microprofile-lra#8]
Emily asked if we could change the name of the terminal() method to terminate():
https://github.com/eclipse/microprofile-lra/blob/master/api/src/main/java/org/eclipse/microprofile/lra/annotation/LRA.java#L138

**Comment by mmusgrov** [comment]
Note that the annotation attributes that determine when the LRA is ended are delayClose and terminal and they are only provided to override the default behaviour.

If there is no context present on entry to the method call and the annotations say to create a new one (ie REQUIRED or REQUIRES_NEW) then, when the method finishes the default is for the spec implementation to automatically close the LRA when the method finishes. To override this behaviour set the delayClose attribute to true.

But if there is a context present on entry to the method call and the annotations say to use that context during the method call (ie REQUIRED or SUPPORTS) then normally the LRA will not be closed when the method ends. To override this behaviour set the terminal attribute to true.

The reason I specified separate annotations is so the defaults make sense:

delayClose on newly created LRAs should default to false
terminate on existing LRAs should default to false

We could use a single attribute but I do not know what name we could use that would capture both meanings (hence I went for two separate annotations to enhance the comprehensibility of the spec).
